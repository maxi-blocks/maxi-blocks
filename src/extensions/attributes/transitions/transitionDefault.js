const transitionDefault = {
	canvas: {
		border: {
			title: 'Border',
			target: ['', ' > .maxi-background-displayer'],
			property: ['border', 'border-radius', 'top', 'left'],
			hoverProp: 'border-status-hover',
		},
		'box shadow': {
			title: 'Box shadow',
			target: '',
			property: 'box-shadow',
			hoverProp: 'box-shadow-status-hover',
		},
		'background / layer': {
			title: 'Background / Layer',
			target: ' > .maxi-background-displayer > div',
			property: false,
			hoverProp: 'block-background-status-hover',
		},
		opacity: {
			title: 'Opacity',
			target: '',
			property: 'opacity',
			hoverProp: 'opacity-status-hover',
		},
	},
};

export default transitionDefault;
