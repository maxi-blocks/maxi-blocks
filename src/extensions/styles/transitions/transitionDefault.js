const transitionDefault = {
	canvas: {
		border: {
			title: 'Border',
			target: ['', ' > .maxi-background-displayer'],
			property: ['border', 'top', 'left'],
			hoverProp: 'border-status-hover',
		},
		'box shadow': {
			title: 'Box shadow',
			target: '',
			property: 'box-shadow',
			hoverProp: 'box-shadow-status-hover',
		},
		'background / layer': {
			title: 'Background / Layer',
			target: ' > .maxi-background-displayer > div',
			property: false,
			hoverProp: 'block-background-status-hover',
		},
	},
};

export default transitionDefault;
