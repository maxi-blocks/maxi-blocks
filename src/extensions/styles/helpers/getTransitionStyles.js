/**
 * General
 */
const breakpoints = ['general', 'xxl', 'xl', 'l', 'm', 's', 'xs'];

/**
 * Internal dependencies
 */
import getLastBreakpointAttribute from '../getLastBreakpointAttribute';
import transitionDefault from '../transitions/transitionDefault';

/**
 * External dependencies
 */
import { isArray, isEmpty, isEqual, isNil } from 'lodash';

/**
 * Generates size styles object
 *
 * @param {Object} obj Block size properties
 */
const getTransitionStyles = (props, transitionObj = transitionDefault) => {
	const { transition } = props;

	if (isEmpty(transition)) return null;

	const response = {};

	Object.entries(transitionObj).forEach(([type, obj]) => {
		Object.entries(obj).forEach(([key, value]) => {
			const { hoverProp: rawHoverProp, isTransform = false } = value;
			const hoverProp =
				!rawHoverProp || isArray(rawHoverProp)
					? rawHoverProp
					: [rawHoverProp];
			if (
				hoverProp &&
				hoverProp.every(prop => !props[prop]) &&
				!isTransform
			)
				return;

			const { target: rawTarget, property: rawProperty } = value;
			const targets = isArray(rawTarget) ? rawTarget : [rawTarget];
			const properties = isArray(rawProperty)
				? rawProperty
				: [rawProperty];

			targets.forEach(target => {
				const hoverTarget = `${target}:hover`;

				breakpoints.forEach(breakpoint => {
					/**
					 * Check if there is any hover transform styles.
					 */
					if (
						isTransform &&
						['scale', 'rotate', 'translate', 'origin'].every(
							prop =>
								!getLastBreakpointAttribute({
									target: `transform-${prop}`,
									breakpoint,
									attributes: props,
									keys: [key, 'hover-status'],
								})
						)
					)
						return;

					if (isNil(response[target]))
						response[target] = { transition: {} };

					const generateTransitionString = (
						target,
						isHover = false
					) => {
						const lastTransitionSplit = getLastBreakpointAttribute({
							target: 'split',
							breakpoint,
							attributes: transition[type][key],
						});
						const transitionSplit =
							transition[type][key]?.[`split-${breakpoint}`];

						const isNewTransitionSplit =
							isEqual(transitionSplit, lastTransitionSplit) &&
							!isNil(transitionSplit);

						/**
						 * If no hover styles was created and no new transition split,
						 * then we don't need to generate transition styles for hover
						 * and duplicate normal transition styles for hover.
						 */
						if (isHover && isNil(response[target])) {
							if (!isNewTransitionSplit) return;

							response[target] = { transition: {} };
						}

						const transitionContent =
							isHover || !lastTransitionSplit
								? transition[type][key]
								: transition[type][key]?.out;

						let transitionString = '';

						const getLastTransitionAttribute = target =>
							getLastBreakpointAttribute({
								target,
								breakpoint,
								attributes: transitionContent,
							});

						const getTransitionAttribute = target =>
							transitionContent?.[`${target}-${breakpoint}`];

						const lastTransitionDuration =
							getLastTransitionAttribute('transition-duration');
						const transitionDuration = getTransitionAttribute(
							'transition-duration'
						);

						const lastTransitionDelay =
							getLastTransitionAttribute('transition-delay');
						const transitionDelay =
							getTransitionAttribute('transition-delay');

						const lastTransitionTimingFunction =
							getLastTransitionAttribute('easing');
						const transitionTimingFunction =
							getTransitionAttribute('easing');

						const lastTransitionStatus =
							getLastTransitionAttribute('transition-status');
						const transitionStatus =
							getTransitionAttribute('transition-status');

						const isSomeValue =
							isEqual(
								transitionDuration,
								lastTransitionDuration
							) ||
							isEqual(transitionDelay, lastTransitionDelay) ||
							isEqual(
								transitionTimingFunction,
								lastTransitionTimingFunction
							) ||
							isEqual(transitionStatus, lastTransitionStatus) ||
							(isHover && isNewTransitionSplit);

						if (isSomeValue) {
							properties.forEach(property => {
								const transitionProperty = property || 'all';

								if (!lastTransitionStatus) {
									transitionString += `${transitionProperty} 0s 0s, `;
								} else if (lastTransitionStatus) {
									transitionString += `${transitionProperty} ${lastTransitionDuration}s ${lastTransitionDelay}s ${lastTransitionTimingFunction}, `;
								}
							});
						}

						transitionString = transitionString.replace(
							/,\s*$/,
							''
						);

						if (transitionString)
							if (isNil(response[target].transition[breakpoint]))
								response[target].transition[breakpoint] = {
									transition: transitionString,
								};
							else
								response[target].transition[
									breakpoint
								].transition += `, ${transitionString}`;
					};

					generateTransitionString(target);
					generateTransitionString(hoverTarget, true);
				});
			});
		});
	});

	return response;
};

export default getTransitionStyles;
