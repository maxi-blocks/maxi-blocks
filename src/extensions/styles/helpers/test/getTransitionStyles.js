/**
 * Internal dependencies
 */
import getTransitionStyles from '../getTransitionStyles';
import transitionDefault from '../../transitions/transitionDefault';

/**
 * External dependencies
 */
import { merge } from 'lodash';

describe('getTransitionStyles', () => {
	it('Get a correct transition styles', () => {
		const object = {
			'border-status-hover': true,
			'box-shadow-status-hover': false,
			'block-background-status-hover': true,
			transition: {
				block: {},
				canvas: {
					border: {
						'transition-duration-general': 0.3,
						'transition-delay-general': 0,
						'easing-general': 'ease',
						'transition-status-general': true,
						hoverProp: 'border-status-hover',
					},
					'box shadow': {
						'transition-duration-general': 0.3,
						'transition-delay-general': 0,
						'easing-general': 'ease',
						'transition-status-general': true,
						hoverProp: 'box-shadow-status-hover',
					},
					'background / layer': {
						'transition-duration-general': 0.3,
						'transition-delay-general': 0,
						'easing-general': 'ease',
						'transition-status-general': false,
						hoverProp: 'block-background-status-hover',
					},
				},
			},
		};

		const result = getTransitionStyles(object);
		expect(result).toMatchSnapshot();
	});

	const repeatedAttributes = {
		'transition-duration-general': 0.3,
		'transition-duration-l': 0.4,
		'transition-duration-m': 0.5,
		'transition-duration-s': 0.6,
		'transition-duration-xs': 0.7,
		'transition-duration-xxl': 0.8,
		'transition-delay-general': 0,
		'transition-delay-l': 0.1,
		'transition-delay-m': 0.2,
		'transition-delay-s': 0.3,
		'transition-delay-xs': 0.4,
		'transition-delay-xxl': 0.5,
		'easing-general': 'ease',
		'easing-l': 'ease-in',
		'easing-m': 'ease-out',
		'easing-s': 'ease-in-out',
		'easing-xs': 'linear',
		'easing-xxl': 'ease-in-out',
	};

	const repeatedGeneralAttributes = {
		'transition-duration-general': 0.3,
		'transition-delay-general': 0,
		'easing-general': 'ease',
		'transition-status-general': true,
	};

	it('Get a correct responsive transition styles', () => {
		const object = {
			'border-status-hover': true,
			'box-shadow-status-hover': true,
			'block-background-status-hover': false,
			transition: {
				block: {},
				canvas: {
					border: {
						...repeatedAttributes,
						'transition-status-general': true,
						'transition-status-l': false,
						'transition-status-m': true,
						'transition-status-s': false,
						'transition-status-xs': true,
						'transition-status-xxl': false,
						hoverProp: 'border-status-hover',
					},
					'box shadow': {
						...repeatedAttributes,
						'transition-status-general': false,
						'transition-status-l': true,
						'transition-status-m': false,
						'transition-status-s': true,
						'transition-status-xs': false,
						'transition-status-xxl': true,
						hoverProp: 'box-shadow-status-hover',
					},
					'background / layer': {
						...repeatedAttributes,
						'transition-status-general': true,
						'transition-status-l': false,
						'transition-status-m': true,
						'transition-status-s': false,
						'transition-status-xs': true,
						'transition-status-xxl': false,
						hoverProp: 'block-background-status-hover',
					},
				},
			},
		};

		const result = getTransitionStyles(object);
		expect(result).toMatchSnapshot();
	});

	const customTransitionObj = {
		...transitionDefault,
		block: {
			typography: {
				title: 'Typography',
				target: ' .maxi-button-block__content',
				property: false,
				hoverProp: 'typography-status-hover',
			},
			'button background': {
				title: 'Button background',
				target: ' .maxi-button-block__button',
				property: 'background',
				hoverProp: 'button-background-status-hover',
			},
		},
	};

	const repeatedAttributesWithStatus = {
		...repeatedAttributes,
		'transition-status-general': true,
		'transition-status-l': false,
		'transition-status-m': true,
		'transition-status-s': false,
		'transition-status-xs': true,
		'transition-status-xxl': false,
	};

	const responsiveTransitionAttributes = {
		'border-status-hover': true,
		'box-shadow-status-hover': true,
		'block-background-status-hover': false,
		'typography-status-hover': true,
		'button-background-status-hover': false,
		transition: {
			block: {
				typography: repeatedAttributesWithStatus,
				'button background': repeatedAttributesWithStatus,
			},
			canvas: {
				border: repeatedAttributesWithStatus,
				'box shadow': repeatedAttributesWithStatus,
				'background / layer': repeatedAttributesWithStatus,
			},
		},
	};

	it('Get a correct responsive transition styles with custom transitionObj', () => {
		const result = getTransitionStyles(
			responsiveTransitionAttributes,
			customTransitionObj
		);
		expect(result).toMatchSnapshot();
	});

	it('Get a correct in/out transition styles', () => {
		const object = {
			'border-status-hover': true,
			'box-shadow-status-hover': true,
			'block-background-status-hover': true,
			transition: {
				block: {},
				canvas: {
					border: {
						'transition-duration-general': 0.3,
						'transition-delay-general': 0,
						'easing-general': 'ease',
						'transition-status-general': true,
						out: {
							'transition-duration-general': 0.9,
							'transition-delay-general': 0.2,
							'easing-general': 'ease-out',
							'transition-status-general': true,
						},
						'split-general': true,
						hoverProp: 'border-status-hover',
					},
					'box shadow': {
						'transition-duration-general': 0.3,
						'transition-delay-general': 0,
						'easing-general': 'ease',
						'transition-status-general': true,
						out: {
							'transition-duration-general': 0.9,
							'transition-delay-general': 0.2,
							'easing-general': 'ease-out',
							'transition-status-general': true,
						},
						'split-general': false,
						hoverProp: 'box-shadow-status-hover',
					},
					'background / layer': {
						'transition-duration-general': 0.3,
						'transition-delay-general': 0,
						'easing-general': 'ease',
						'transition-status-general': false,
						hoverProp: 'block-background-status-hover',
					},
				},
			},
		};

		const result = getTransitionStyles(object);
		expect(result).toMatchSnapshot();
	});

	it('Get a correct responsive in/out transition styles', () => {
		const repeatedOutAttributes = {
			'transition-duration-general': 1.8,
			'transition-duration-l': 1.7,
			'transition-duration-m': 1.6,
			'transition-duration-s': 1.5,
			'transition-duration-xs': 1.4,
			'transition-duration-xxl': 1.3,
			'transition-delay-general': 1,
			'transition-delay-l': 0.9,
			'transition-delay-m': 0.8,
			'transition-delay-s': 0.7,
			'transition-delay-xs': 0.6,
			'transition-delay-xxl': 0.5,
			'easing-general': 'ease-in-out',
			'easing-l': 'ease-out',
			'easing-m': 'ease-in',
			'easing-s': 'linear',
			'easing-xs': 'ease-in-out',
			'easing-xxl': 'linear',
			'transition-status-general': true,
		};

		const object = {
			'border-status-hover': true,
			'box-shadow-status-hover': true,
			'block-background-status-hover': false,
			transition: {
				block: {},
				canvas: {
					border: {
						...repeatedAttributes,
						out: repeatedOutAttributes,
						'transition-status-general': true,
						'transition-status-l': false,
						'transition-status-m': true,
						'transition-status-s': false,
						'transition-status-xs': true,
						'transition-status-xxl': false,
						'split-general': true,
						'split-l': true,
						'split-m': false,
						'split-s': true,
						'split-xs': true,
						'split-xxl': false,
						hoverProp: 'border-status-hover',
					},
					'box shadow': {
						...repeatedAttributes,
						out: repeatedOutAttributes,
						'transition-status-general': false,
						'transition-status-l': true,
						'transition-status-m': false,
						'transition-status-s': true,
						'transition-status-xs': false,
						'transition-status-xxl': true,
						'split-general': false,
						'split-l': false,
						'split-m': true,
						'split-s': true,
						'split-xs': false,
						'split-xxl': true,
						hoverProp: 'box-shadow-status-hover',
					},
					'background / layer': {
						...repeatedAttributes,
						out: repeatedOutAttributes,
						'transition-status-general': true,
						'transition-status-l': false,
						'transition-status-m': true,
						'transition-status-s': false,
						'transition-status-xs': true,
						'transition-status-xxl': false,
						'split-general': true,
						'split-l': true,
						'split-m': false,
						'split-s': true,
						'split-xs': true,
						'split-xxl': false,
						hoverProp: 'block-background-status-hover',
					},
				},
			},
		};

		const result = getTransitionStyles(object);
		expect(result).toMatchSnapshot();
	});

	const transformTransitionObj = {
		transform: {
			canvas: {
				title: 'Canvas',
				target: '',
				property: 'transform',
				isTransform: true,
			},
			block: {
				title: 'Block',
				target: ' .block-class',
				property: 'transform',
				isTransform: true,
			},
			button: {
				title: 'Button',
				target: ' .button-class',
				property: 'transform',
				isTransform: true,
			},
		},
	};

	it('Get a correct transform transition styles', () => {
		const object = {
			'transform-scale-general': {
				canvas: {
					'hover-status': true,
				},
				block: {
					'hover-status': false,
				},
			},
			'transform-rotate-general': {
				canvas: {
					'hover-status': false,
				},
				block: {
					'hover-status': true,
				},
			},
			transition: {
				transform: {
					canvas: repeatedGeneralAttributes,
					block: repeatedGeneralAttributes,
					button: repeatedGeneralAttributes,
				},
			},
		};

		const result = getTransitionStyles(object, transformTransitionObj);
		expect(result).toMatchSnapshot();
	});

	it('Get a correct responsive transform(mixed with others) transition styles', () => {
		const transitionObj = {
			...customTransitionObj,
			...transformTransitionObj,
		};

		const object = merge(
			{
				'transform-scale-general': {
					canvas: {
						'hover-status': true,
					},
					block: {
						'hover-status': false,
					},
				},
				'transform-rotate-general': {
					canvas: {
						'hover-status': false,
					},
				},
				'transform-rotate-m': {
					block: {
						'hover-status': true,
					},
				},
				'transform-scale-s': {
					canvas: {
						'hover-status': false,
					},
				},
				transition: {
					transform: {
						canvas: repeatedAttributesWithStatus,
						block: repeatedAttributesWithStatus,
						button: repeatedAttributesWithStatus,
					},
				},
			},
			repeatedAttributesWithStatus
		);

		const result = getTransitionStyles(object, transitionObj);
		expect(result).toMatchSnapshot();
	});
});
