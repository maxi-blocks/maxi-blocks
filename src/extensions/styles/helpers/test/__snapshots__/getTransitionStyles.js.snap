// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTransitionStyles Get a correct in/out transition styles 1`] = `
{
  "": {
    "transition": {
      "general": {
        "transition": "border 0.9s 0.2s ease-out, border-radius 0.9s 0.2s ease-out, top 0.9s 0.2s ease-out, left 0.9s 0.2s ease-out, box-shadow 0.3s 0s ease",
      },
    },
  },
  " > .maxi-background-displayer": {
    "transition": {
      "general": {
        "transition": "border 0.9s 0.2s ease-out, border-radius 0.9s 0.2s ease-out, top 0.9s 0.2s ease-out, left 0.9s 0.2s ease-out",
      },
    },
  },
  " > .maxi-background-displayer > div": {
    "transition": {
      "general": {
        "transition": "all 0s 0s",
      },
    },
  },
  " > .maxi-background-displayer > div > svg": {
    "transition": {
      "general": {
        "transition": "all 0s 0s",
      },
    },
  },
  " > .maxi-background-displayer:hover": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease",
      },
    },
  },
  ":hover": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease, box-shadow 0.3s 0s ease",
      },
    },
  },
}
`;

exports[`getTransitionStyles Get a correct responsive in/out transition styles 1`] = `
{
  "": {
    "transition": {
      "general": {
        "transition": "border 1.8s 1s ease-in-out, border-radius 1.8s 1s ease-in-out, top 1.8s 1s ease-in-out, left 1.8s 1s ease-in-out, box-shadow 0s 0s",
      },
      "l": {
        "transition": "border 1.7s 0.9s ease-out, border-radius 1.7s 0.9s ease-out, top 1.7s 0.9s ease-out, left 1.7s 0.9s ease-out, box-shadow 0.4s 0.1s ease-in",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out, box-shadow 1.6s 0.8s ease-in",
      },
      "s": {
        "transition": "border 1.5s 0.7s linear, border-radius 1.5s 0.7s linear, top 1.5s 0.7s linear, left 1.5s 0.7s linear, box-shadow 1.5s 0.7s linear",
      },
      "xs": {
        "transition": "border 1.4s 0.6s ease-in-out, border-radius 1.4s 0.6s ease-in-out, top 1.4s 0.6s ease-in-out, left 1.4s 0.6s ease-in-out, box-shadow 0s 0s",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 1.3s 0.5s linear",
      },
    },
  },
  " > .maxi-background-displayer": {
    "transition": {
      "general": {
        "transition": "border 1.8s 1s ease-in-out, border-radius 1.8s 1s ease-in-out, top 1.8s 1s ease-in-out, left 1.8s 1s ease-in-out",
      },
      "l": {
        "transition": "border 1.7s 0.9s ease-out, border-radius 1.7s 0.9s ease-out, top 1.7s 0.9s ease-out, left 1.7s 0.9s ease-out",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out",
      },
      "s": {
        "transition": "border 1.5s 0.7s linear, border-radius 1.5s 0.7s linear, top 1.5s 0.7s linear, left 1.5s 0.7s linear",
      },
      "xs": {
        "transition": "border 1.4s 0.6s ease-in-out, border-radius 1.4s 0.6s ease-in-out, top 1.4s 0.6s ease-in-out, left 1.4s 0.6s ease-in-out",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
    },
  },
  " > .maxi-background-displayer:hover": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease",
      },
      "l": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out",
      },
      "s": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
      "xs": {
        "transition": "border 0.7s 0.4s linear, border-radius 0.7s 0.4s linear, top 0.7s 0.4s linear, left 0.7s 0.4s linear",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
    },
  },
  ":hover": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease, box-shadow 0s 0s",
      },
      "l": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0.4s 0.1s ease-in",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out, box-shadow 0s 0s",
      },
      "s": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0.6s 0.3s ease-in-out",
      },
      "xs": {
        "transition": "border 0.7s 0.4s linear, border-radius 0.7s 0.4s linear, top 0.7s 0.4s linear, left 0.7s 0.4s linear, box-shadow 0s 0s",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0.8s 0.5s ease-in-out",
      },
    },
  },
}
`;

exports[`getTransitionStyles Get a correct responsive transform(mixed with others) transition styles 1`] = `
{
  "": {
    "transition": {
      "general": {
        "transition": "transform 0.3s 0s ease",
      },
      "l": {
        "transition": "transform 0s 0s",
      },
      "m": {
        "transition": "transform 0.5s 0.2s ease-out",
      },
      "xxl": {
        "transition": "transform 0s 0s",
      },
    },
  },
  " .block-class": {
    "transition": {
      "m": {
        "transition": "transform 0.5s 0.2s ease-out",
      },
      "s": {
        "transition": "transform 0s 0s",
      },
      "xs": {
        "transition": "transform 0.7s 0.4s linear",
      },
    },
  },
}
`;

exports[`getTransitionStyles Get a correct responsive transition styles 1`] = `
{
  "": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease, box-shadow 0s 0s",
      },
      "l": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0.4s 0.1s ease-in",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out, box-shadow 0s 0s",
      },
      "s": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0.6s 0.3s ease-in-out",
      },
      "xs": {
        "transition": "border 0.7s 0.4s linear, border-radius 0.7s 0.4s linear, top 0.7s 0.4s linear, left 0.7s 0.4s linear, box-shadow 0s 0s",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0.8s 0.5s ease-in-out",
      },
    },
  },
  " > .maxi-background-displayer": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease",
      },
      "l": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out",
      },
      "s": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
      "xs": {
        "transition": "border 0.7s 0.4s linear, border-radius 0.7s 0.4s linear, top 0.7s 0.4s linear, left 0.7s 0.4s linear",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
    },
  },
}
`;

exports[`getTransitionStyles Get a correct responsive transition styles with custom transitionObj 1`] = `
{
  "": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease, box-shadow 0.3s 0s ease",
      },
      "l": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0s 0s",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out, box-shadow 0.5s 0.2s ease-out",
      },
      "s": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0s 0s",
      },
      "xs": {
        "transition": "border 0.7s 0.4s linear, border-radius 0.7s 0.4s linear, top 0.7s 0.4s linear, left 0.7s 0.4s linear, box-shadow 0.7s 0.4s linear",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s, box-shadow 0s 0s",
      },
    },
  },
  " .maxi-button-block__content": {
    "transition": {
      "general": {
        "transition": "all 0.3s 0s ease",
      },
      "l": {
        "transition": "all 0s 0s",
      },
      "m": {
        "transition": "all 0.5s 0.2s ease-out",
      },
      "s": {
        "transition": "all 0s 0s",
      },
      "xs": {
        "transition": "all 0.7s 0.4s linear",
      },
      "xxl": {
        "transition": "all 0s 0s",
      },
    },
  },
  " > .maxi-background-displayer": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease",
      },
      "l": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
      "m": {
        "transition": "border 0.5s 0.2s ease-out, border-radius 0.5s 0.2s ease-out, top 0.5s 0.2s ease-out, left 0.5s 0.2s ease-out",
      },
      "s": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
      "xs": {
        "transition": "border 0.7s 0.4s linear, border-radius 0.7s 0.4s linear, top 0.7s 0.4s linear, left 0.7s 0.4s linear",
      },
      "xxl": {
        "transition": "border 0s 0s, border-radius 0s 0s, top 0s 0s, left 0s 0s",
      },
    },
  },
}
`;

exports[`getTransitionStyles Get a correct transform transition styles 1`] = `
{
  "": {
    "transition": {
      "general": {
        "transition": "transform 0.3s 0s ease",
      },
    },
  },
  " .block-class": {
    "transition": {
      "general": {
        "transition": "transform 0.3s 0s ease",
      },
    },
  },
}
`;

exports[`getTransitionStyles Get a correct transition styles 1`] = `
{
  "": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease",
      },
    },
  },
  " > .maxi-background-displayer": {
    "transition": {
      "general": {
        "transition": "border 0.3s 0s ease, border-radius 0.3s 0s ease, top 0.3s 0s ease, left 0.3s 0s ease",
      },
    },
  },
  " > .maxi-background-displayer > div": {
    "transition": {
      "general": {
        "transition": "all 0s 0s",
      },
    },
  },
  " > .maxi-background-displayer > div > svg": {
    "transition": {
      "general": {
        "transition": "all 0s 0s",
      },
    },
  },
}
`;
